### PR TITLE
Get security notices from the database

### LINKED ISSUE #6322: [Security notices] build main page
[UX document](https://docs.google.com/document/d/183fLUXOLL17_baS617aX7Sj133n4lLfS6-rLhXXv9I0/edit#)

build /security/notices

### LINKED ISSUE #6323: [Security notices] build search results view
[UX document](https://docs.google.com/document/d/183fLUXOLL17_baS617aX7Sj133n4lLfS6-rLhXXv9I0/edit#)

build search results variant of main page (/security/notices?release=foo&package=bar&details=hum)

### LINKED ISSUE #6324: [Security notices] build individual notice page
[UX document](https://docs.google.com/document/d/183fLUXOLL17_baS617aX7Sj133n4lLfS6-rLhXXv9I0/edit#)

build /security/notices/{id}

### PR DESCRIPTION
**Pull request by jpmartinspt**
## Done

Implement views to fetch security notices from a database.
This work includes adding all the necessary boilerplate to connect to the database.

## QA

- Check out this feature branch
- Run the site using the command `./run serve`
- View the site locally in your web browser at: http://0.0.0.0:8001/security/notices
- Run through the following [QA steps](https://canonical-web-and-design.github.io/practices/workflow/qa-steps.html)

## Issue / Card

Fixes #6324
Fixes #6322 
Fixes #6323 
Fixes #2196

### CODE REVIEW

**Comment by anasereijo** [comment]
### /security/notices
Hero:

- [ ] title should be `h1`

- [ ] can we make the text `col-8`?

- [ ] RSS and mailing list are not working when I click?

Search:

- [ ] can the search button be disabled until the user types or picks a filter? I think it's how Matthew's UX was designed

### Search results

- [ ] can you remove the padding (or marging not sure) from the filter? see below
Screenshot 2020-02-10 at 10 11 05

- [ ] mobile: hide `hr` on mobile search results. It's just taking space

Screenshot 2020-02-10 at 10 15 12


### /notices

- [ ] title should be `h1`

**Comment by jpmartinspt** [comment]
@anasereijo updated with your input. RSS and the mailing list are not working yet, should be updated in a future PR.

**Comment by anasereijo** [comment]
- [ ] Search results title should be `h2` as it was before

**Comment by anasereijo** [comment]
🥇 awesome joao!

File: templates/security/notices.html [hunk 1]
```diff
@@ -1 +1,142 @@
...
\ No newline at end of file
+{% extends "templates/one-column.html" %}
+{% set search = request.args.get('release') or request.args.get('details') %}
+{% block title %}Security notices{% endblock %}
+{% block content %}
+<section class="p-strip--suru-topped">
+  <div class="row">
+    <div class="col-12">
+      {% if search %}
+      <h2>Ubuntu Security Notices - Search Results</h2>
```
Comment (sowasred2012) [inline templates/security/notices.html]: This should also be a `h1`:
```suggestion
      <h1>Ubuntu Security Notices - Search Results</h1>
```

File: templates/security/_notice-brief.html [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<article class="notice" style="padding-bottom: 32px;">
+  <h4 class="u-no-margin" style="padding-bottom: 8px;"><a href="/security/notices/{{notice.id}}">{{ notice.id }}: {{ notice.title }} ›</a></h4>
```
Comment (sowasred2012) [inline templates/security/_notice-brief.html]:
```suggestion
  <h3 class="u-no-margin p-heading--four" style="padding-bottom: 8px;"><a href="/security/notices/{{notice.id}}">{{ notice.id }}: {{ notice.title }} ›</a></h3>
```

File: templates/security/notices.html (hunk 1 above)
Comment (jpmartinspt) [reply templates/security/notices.html]: I think we want it as an `h2`
https://github.com/canonical-web-and-design/ubuntu.com/pull/6420#issuecomment-584080664.

Comment (sowasred2012) [reply templates/security/notices.html]: In that case, `<h1 class="p-heading--two">`, the page should have a `h1`.

**Comment by jpmartinspt** [comment]
@sowasred2012 suggestions merged, just learned why do we use those classes, cheers!

File: templates/security/notices.html [hunk 3]
```diff
@@ -1 +1,142 @@
...
+
+  // Disable search when form is empty
+  function toogleSearch() {
+    submit.disabled = !details.value && !release.value;
+  }
+
+  form.onchange = function(event) {
+    toogleSearch();
+  }
+
```
Comment (sowasred2012) [inline templates/security/notices.html]: Sorry, one more suggestion - first thing I did was type something into the `#details` input, but the submit button didn't activate until I'd clicked somewhere else on the page. You could also attach a `keyup` event:

```suggestion

  details.onkeyup = function() {
    toogleSearch();
  }
  
```

Comment (jpmartinspt) [reply templates/security/notices.html]: Added

**Comment by cassiocassio** [comment]
### /security/notices

**Comment by cassiocassio** [comment]
all non-blocking
The first one is my strongest opinion.

### /security/notices

- [ ] In Search / release menu Is it possible we can show LTS for the releases that are ? Thats standard in a lot of places "18.04 LTS"

- [ ] The label "Details contain", or perhaps the search field hint text, could explain you can search for CVE, package, USN or details. Either it's just a blank google-like field, and they assume it is magic, or we say what the details are...

- [ ] the right-left alignment of the search button, the subscibe input field, and the underlying grid makes me nervous. I think it's built as designed, but a bit unhappy. 

### /notices

- [ ] Did we ever consider the References CVEs being further up the page, being on the right on wide monitors, or generally visible without scrolling? Is a user need for a USN "Does this fix that vulnerability?"

### searchresults

- [ ] Super theoretical: future work : if i really cared about a particular vulnerability, and I got zero search results, I would totally provide me email and ask for an alert when it's fixed. Or an RSS feed. Or anything.

**Comment by jpmartinspt** [comment]
@cassiocassio addressed a couple of your comments, would you be willing to UX +1 this one and we could raise other tickets for things coming out of the discussion we are planning to have?

**Comment by matthewpaulthomas** [comment]
> - The label "Details contain", or perhaps the search field hint text, could explain you can search for CVE, package, USN or details. Either it's just a blank google-like field, and they assume it is magic, or we say what the details are...

“Details” is the heading of the section that this field searches. I don’t think we should relabel the field unless we rename the section too (and I think the section heading is reasonable).

> - the right-left alignment of the search button, the subscibe input field, and the underlying grid makes me nervous. I think it's built as designed, but a bit unhappy.

I guess there’s a maximum useful width for the search field, beyond which the Search button would no longer touch the right edge of the grid. But with a maximised window on a 1366 × 768 display I don’t come close to reaching that maximum width.

> - Did we ever consider the References CVEs being further up the page, being on the right on wide monitors, or generally visible without scrolling? Is a user need for a USN "Does this fix that vulnerability?"

In the design spec I wrote: “Now that CVEs are being linked to in context, an out-of-context list of CVEs is no longer any use, so the ‘References’ section should be dropped.” Unfortunately neither of those things is happening yet: CVE IDs are unhelpfully unlinked, and so the References section is still present.

> - Super theoretical: future work : if i really cared about a particular vulnerability, and I got zero search results, I would totally provide me email and ask for an alert when it's fixed. Or an RSS feed. Or anything.

Yes, it would be nice if you could subscribe to a search regardless of whether it currently has zero results. The results page would have a “Subscribe” box the same as the main page does, just with a different heading.

**Comment by matthewpaulthomas** [comment]
UX–1 because:
- [ ] As mentioned above, CVE IDs in the Details aren’t being auto-linked. (I recognise that there’s ongoing discussion about whether the “References” section is really needed, so I won’t insist on dropping it.)
- [ ] Details paragraphs are being munged together. Compare [the two paragraphs in USN-4196-1’s Markdown](https://git.launchpad.net/usn.ubuntu.com/tree/content/4196-1.md#n32) with [the one paragraph in the output](https://ubuntu-com-canonical-web-and-design-pr-6420.run.demo.haus/security/notices/4197-1).

**Comment by matthewpaulthomas** [comment]
From discussion with João, I’m happy for this to be UX+1 if separate issues are reported for those three items, and those issues block public release.

**Comment by jpmartinspt** [comment]
Merged and I'll make sure I raise the issues.

**Pull request merged**